### PR TITLE
core: add a combat log event system

### PR DESCRIPTION
**Pull request by Rainrider**
Will probably not see much use, but won't hurt either.

### CODE REVIEW

**Review by p3lim — APPROVED** [review]
Only comment I have on this is that if layouts start mis-using this it'll show blame on oUF for being resource-hungry. A _very_ minor remark, the PR looks good otherwise, but I haven't tested it myself.

**Comment by ls-** [comment]
I like the idea as well, I'm going to test this PR later this week, have my hands full atm. I have a floating combat text plug-in that uses CLEU, I'll try rewriting with this PR in place to see what's missing and what I'd like to be added/removed/changed.

File: combatevents.lua [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+		for _, func in next, funcs do
+			func(self, ...)
+		end
+	end
+}
+
+local listener = CreateFrame('Frame')
+listener.activeEvents = 0
+
+local function filter(_, event, ...)
```
Comment (p3lim) [inline combatevents.lua:31]: I'd expand on the name of this function, processEvent perhaps?

Comment (Rainrider) [reply combatevents.lua:31]: It actually filters the events and handles the processing to the registered listener. Event is here the combat log event, maybe I should rather rename this?

**Pull request merged**